store/src/pmmr.rs

```diff
@@ -57,6 +57,7 @@ where
 	T: PMMRable,
 {
 	data_dir: String,
+	prunable: bool,
```

Comment:
:+1:

I suspect we can take advantage of this `prunable` bool in a subsequent PR and eliminate the `pmmr_leaf.bin` and `pmmr_prun.bin` files completely.

Reply:
I suspect so too, or maybe layer the pruning part of the PMMR backend on top of the regular one.